crates/modelardb_server/src/query/mod.rs

```diff
@@ -485,13 +489,24 @@ impl TableProvider for ModelTable {
         let (maybe_parquet_predicates, maybe_grid_predicates) =
             rewrite_and_combine_filters(schema, filters);
 
-        // Compute a mapping from hashes to the requested tag values in the requested order.
-        let hash_to_tags = self
-            .context
-            .table_metadata_manager
-            .mapping_from_hash_to_tags(table_name, &stored_tag_columns_in_projection)
-            .await
-            .map_err(|error| DataFusionError::Plan(error.to_string()))?;
+        // Compute a mapping from hashes to the requested tag values in the requested order. If the
+        // server is a cloud node, use the table metadata manager for the remote metadata database.
+        let configuration_manager = self.context.configuration_manager.read().await;
+        let hash_to_tags = if let (ServerMode::Cloud, ClusterMode::MultiNode(manager)) = (
```

Comment:
I have attempted this a few times but I am not sure how it would work. It is my understanding that Rust generics are set during compile-time so having a function that returns two different types based on run-time might not work.

Reply:
I was thinking that maybe it would be possible to create such a function by returning `&TableMetadataManager<dyn Database>`, so by using a combination of generic and traits. However, to do so a `where` constraint with the traits that are part of the [`Database`](https://docs.rs/sqlx/latest/sqlx/trait.Database.html) trait and I am not sure it can be defined without including `SQLite` or `PostgreSQL` in the constraint.

Reply:
I have attempted that as well a few times without success. Lets leave it as it is for now and then we can always come back to it if we get any good ideas.